dp_mobility_report/model/preprocessing.py

```diff
@@ -58,40 +59,61 @@ def preprocess_data(
     tessellation: gpd.GeoDataFrame,
     max_trips_per_user: int,
     user_privacy: bool,
+    timestamps: bool
 ) -> pd.DataFrame:
-    df = _validate_columns(df)
+    df = _validate_columns(df, timestamps)
 
     df.loc[:, const.ID] = range(0, len(df))
 
     # make sure trip ids are unique and ordered correctly
-    df[const.TID] = (
-        df.sort_values([const.UID, const.DATETIME])
-        .groupby([const.UID, const.TID], sort=False)
-        .ngroup()
-    )
+    if timestamps:
+        df[const.TID] = (
+            df.sort_values([const.UID, const.DATETIME])
+            .groupby([const.UID, const.TID], sort=False)
+            .ngroup()
+        )
 
-    # remove unnessessary columns
-    columns = [const.ID, const.UID, const.TID, const.DATETIME, const.LAT, const.LNG]
-    if const.TILE_ID in df.columns:
-        columns.append(const.TILE_ID)
-    df = df.loc[:, columns]
+        # remove unnessessary columns
+        columns = [const.ID, const.UID, const.TID, const.DATETIME, const.LAT, const.LNG]
+        if const.TILE_ID in df.columns:
+            columns.append(const.TILE_ID)
+        df = df.loc[:, columns]
+        df[const.DATETIME] = pd.to_datetime(df[const.DATETIME])
+        # create time related variables
+        df.loc[:, const.HOUR] = df[const.DATETIME].dt.hour
+        df.loc[:, const.IS_WEEKEND] = np.select(
+            [df[const.DATETIME].dt.weekday > 4], ["weekend"], default="weekday"
+        )
 
-    # create time related variables
-    df.loc[:, const.HOUR] = df[const.DATETIME].dt.hour
-    df.loc[:, const.IS_WEEKEND] = np.select(
-        [df[const.DATETIME].dt.weekday > 4], ["weekend"], default="weekday"
-    )
+        # remove waypoints
+        df = df.sort_values(const.DATETIME).groupby(const.TID).nth([0, -1])
+        df.reset_index(inplace=True)
 
-    # remove waypoints
-    df = df.sort_values(const.DATETIME).groupby(const.TID).nth([0, -1])
-    df.reset_index(inplace=True)
+        # assign start and end as point_type
+        df[const.POINT_TYPE] = "start"
+        df.sort_values(const.DATETIME, inplace=True)
+        df.loc[
+            df.groupby(const.TID)[const.POINT_TYPE].tail(1).index, const.POINT_TYPE
+        ] = "end"
+    else:
+        # make sure trip ids are unique and ordered correctly
```

Comment:
how do we know the correct order of records if there is no datetime? do we need another column "sequence" instead?